src/test/java/org/elasticsearch/cluster/block/ClusterBlockTests.java

```diff
@@ -0,0 +1,85 @@
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.cluster.block;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.common.io.stream.BytesStreamInput;
+import org.elasticsearch.common.io.stream.BytesStreamOutput;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.test.ElasticsearchTestCase;
+import org.junit.Test;
+
+import java.util.EnumSet;
+
+import static org.elasticsearch.cluster.block.ClusterBlockLevel.*;
+import static org.hamcrest.CoreMatchers.equalTo;
+
+public class ClusterBlockTests extends ElasticsearchTestCase {
+
+    private static ClusterBlockLevel[] ALL_CLUSTER_BLOCK_LEVELS = ClusterBlockLevel.ALL.toArray(new ClusterBlockLevel[ALL.size()]);
+
+    @Test
+    public void testSerialization() throws Exception {
+        int iterations = randomIntBetween(5, 20);
+        for (int i = 0; i < iterations; i++) {
+            // Get a random version
+            Version fromVersion = randomVersion();
+            Version toVersion = randomVersion();
+
+            ClusterBlockLevel[] blockLevels = ALL_CLUSTER_BLOCK_LEVELS;
+            if (fromVersion.before(Version.V_1_6_0)) {
+                blockLevels = new ClusterBlockLevel[]{READ, WRITE, METADATA};
+            }
+
+            // Get a random list of ClusterBlockLevels
+            EnumSet<ClusterBlockLevel> levels = EnumSet.noneOf(ClusterBlockLevel.class);
+            int nbLevels = randomIntBetween(1, blockLevels.length);
+            for (int j = 0; j < nbLevels; j++) {
+                levels.add(randomFrom(blockLevels));
+            }
+
+            ClusterBlock clusterBlock = new ClusterBlock(randomInt(), "cluster block #" + randomInt(), randomBoolean(),
+                    randomBoolean(), randomFrom(RestStatus.values()), levels);
+
+            BytesStreamOutput out = new BytesStreamOutput();
+            out.setVersion(toVersion);
+            clusterBlock.writeTo(out);
+
+            BytesStreamInput in = new BytesStreamInput(out.bytes());
+            in.setVersion(toVersion);
+            ClusterBlock result = ClusterBlock.readClusterBlock(in);
+
+            assertThat(result.id(), equalTo(clusterBlock.id()));
+            assertThat(result.status(), equalTo(clusterBlock.status()));
+            assertThat(result.description(), equalTo(clusterBlock.description()));
+            assertThat(result.retryable(), equalTo(clusterBlock.retryable()));
+            assertThat(result.disableStatePersistence(), equalTo(clusterBlock.disableStatePersistence()));
+
+            for (ClusterBlockLevel level : clusterBlock.levels()) {
```

Comment:
I think we cannot check this equality because `METADATA` is deserialized as `METADATA_WRITE + METADATA_READ`.